nav2_bringup/launch/tb3_gz_simulation_launch.py

```diff
@@ -0,0 +1,302 @@
+    )
+
+    declare_use_robot_state_pub_cmd = DeclareLaunchArgument(
+        'use_robot_state_pub',
+        default_value='True',
+        description='Whether to start the robot state publisher',
+    )
+
+    declare_use_rviz_cmd = DeclareLaunchArgument(
+        'use_rviz', default_value='True', description='Whether to start RVIZ'
+    )
+
+    declare_simulator_cmd = DeclareLaunchArgument(
+        'headless', default_value='True', description='Whether to execute gzclient)'
+    )
+
+    declare_world_cmd = DeclareLaunchArgument(
+        'world',
+        # TODO(orduno) Switch back once ROS argument passing has been fixed upstream
+        #              https://github.com/ROBOTIS-GIT/turtlebot3_simulations/issues/91
+        # default_value=os.path.join(get_package_share_directory('turtlebot3_gazebo'),
+        # worlds/turtlebot3_worlds/waffle.model')
+        default_value=os.path.join(bringup_dir, 'worlds', 'gz_world_only.sdf.xacro'),
+        description='Full path to world model file to load',
+    )
+
+    declare_robot_name_cmd = DeclareLaunchArgument(
+        'robot_name', default_value='turtlebot3_waffle', description='name of the robot'
+    )
+
+    declare_robot_sdf_cmd = DeclareLaunchArgument(
+        'robot_sdf',
+        default_value=os.path.join(bringup_dir, 'worlds', 'gz_waffle.sdf'),
+        description='Full path to robot sdf file to spawn the robot in gazebo',
+    )
+
+    urdf = os.path.join(bringup_dir, 'urdf', 'turtlebot3_waffle.urdf')
+    with open(urdf, 'r') as infp:
+        robot_description = infp.read()
+
+    start_robot_state_publisher_cmd = Node(
+        condition=IfCondition(use_robot_state_pub),
+        package='robot_state_publisher',
+        executable='robot_state_publisher',
+        name='robot_state_publisher',
+        namespace=namespace,
+        output='screen',
+        parameters=[
+            {'use_sim_time': use_sim_time, 'robot_description': robot_description}
+        ],
+        remappings=remappings,
+    )
+
+    rviz_cmd = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(os.path.join(launch_dir, 'rviz_launch.py')),
+        condition=IfCondition(use_rviz),
+        launch_arguments={
+            'namespace': namespace,
+            'use_namespace': use_namespace,
+            'rviz_config': rviz_config_file,
+        }.items(),
+    )
+
+    bringup_cmd = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(os.path.join(launch_dir, 'bringup_launch.py')),
+        launch_arguments={
+            'namespace': namespace,
+            'use_namespace': use_namespace,
+            'slam': slam,
+            'map': map_yaml_file,
+            'use_sim_time': use_sim_time,
+            'params_file': params_file,
+            'autostart': autostart,
+            'use_composition': use_composition,
+            'use_respawn': use_respawn,
+        }.items(),
+    )
+
+    world_sdf = tempfile.mktemp(prefix='nav2_', suffix='.sdf')
```

Comment:
What's going on here?

Reply:
The SDF file for the world is a xacro file because we wanted to conditionally load the `SceneBroadcaster` plugin based on wheter we're running in headless mode. But currently, the Gazebo command line doesn't take SDF strings for worlds, so the output of `xacro` needs to be saved into a temporary file and passed to Gazebo. This was added to address https://github.com/ros-planning/navigation2/pull/3634#issuecomment-1877822806

Reply:
Please add a comment in the code explaining this

Reply:
Added in https://github.com/ros-planning/navigation2/pull/3634/commits/225f1f3154624d447d80fd7e053aba519a49bb39